src/ResponseFetcher/DefaultResponseFetcher.php

```diff
@@ -39,13 +40,23 @@ public function fetchResponses(RequestInterface $frontend_request, RequestInterf
       'allow_redirects' => FALSE,
       'http_errors' => FALSE,
     ]);
-    $frontend_response = $client->sendAsync($frontend_request);
+    // Try fetching static html via filesystem
+    $root = getenv('PWD');
+    $site = getenv('SITE');
+    $request_body = file_get_contents($root . '/frontend/public/' . $site . '/layout--default.html');
+    if ($request_body) {
+      $frontend_response = new Response(200,[],$request_body);
+    }
+    else {
+      // Fallback to letting server handle the request
+      // $frontend_response = $client->sendAsync($frontend_request);
+      // Resolve the frontend promise first since static files should be always
+      // faster than the backend.
+      /** @var \Psr\Http\Message\ResponseInterface $frontend_response */
+      // $frontend_response = $frontend_response->wait();
```

Comment:
Commented out?

Reply:
Just to be sure it is coming from the filesystem instead of making the request (for testing purposes)

Reply:
Why not throw an error in that case?
Or maybe write a log somewhere but still do it with http request for backup.

Reply:
I was hoping to make it more general because I don't know if the file pattern is the same in every project. 
If frontend/public is located somewhere else than in root dir this will not work. 
If we are sure this (or some other more general solution) will work, then I would remove the http request completely